flyway-core/src/main/java/org/flywaydb/core/internal/resolver/shell/ShellMigrationResolver.java

```diff
@@ -0,0 +1,152 @@
+     */
+    private final String shellMigrationSeparator;
+
+    /**
+     * The suffix for shell migrations
+     */
+    private final String shellMigrationSuffix;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param classLoader             The ClassLoader for loading migrations on the classpath.
+     * @param location                The location on the classpath where to migrations are located.
+     * @param shellMigrationPrefix    The prefix for shell migrations
+     * @param shellMigrationSeparator The separator for shell migrations
+     * @param shellMigrationSuffix    The suffix for shell migrations
+     */
+    public ShellMigrationResolver(ClassLoader classLoader, Location location, String shellMigrationPrefix,
+                                  String shellMigrationSeparator, String shellMigrationSuffix) {
+        this.scanner = new Scanner(classLoader);
+        this.location = location;
+        this.shellMigrationPrefix = shellMigrationPrefix;
+        this.shellMigrationSeparator = shellMigrationSeparator;
+        this.shellMigrationSuffix = shellMigrationSuffix;
+    }
+
+    public List<ResolvedMigration> resolveMigrations() {
+        List<ResolvedMigration> migrations = new ArrayList<ResolvedMigration>();
+
+        Resource[] resources = scanner.scanForResources(location, shellMigrationPrefix, shellMigrationSuffix);
+        for (Resource resource : resources) {
+            ResolvedMigrationImpl resolvedMigration = extractMigrationInfo(resource);
+            resolvedMigration.setPhysicalLocation(resource.getLocationOnDisk());
+            resolvedMigration.setExecutor(new ShellMigrationExecutor(resource));
+
+            migrations.add(resolvedMigration);
+        }
+
+        Collections.sort(migrations, new ResolvedMigrationComparator());
+        return migrations;
+    }
+
+    /**
+     * Extracts the migration info for this resource.
+     *
+     * @param resource The resource to analyse.
+     * @return The migration info.
+     */
+    private ResolvedMigrationImpl extractMigrationInfo(Resource resource) {
+        ResolvedMigrationImpl migration = new ResolvedMigrationImpl();
+
+        Pair<MigrationVersion, String> info =
+                MigrationInfoHelper.extractVersionAndDescription(resource.getFilename(),
+                        shellMigrationPrefix, shellMigrationSeparator, shellMigrationSuffix);
+        migration.setVersion(info.getLeft());
+        migration.setDescription(info.getRight());
+
+        migration.setScript(extractScriptName(resource));
+
+        migration.setChecksum(calculateChecksum(resource.loadAsBytes()));
+        migration.setType(MigrationType.SQL);
+        return migration;
+    }
+
+    /**
+     * Extracts the script name from this resource.
+     *
+     * @param resource The resource to process.
+     * @return The script name.
+     */
+    /* private -> for testing */ String extractScriptName(Resource resource) {
```

Comment:
Yeah, that comment appears often elsewhere in the codebase, and is there specifically having copied ShellMigrationResolver from SqlMigrationResolver, i.e. https://github.com/rollbar/flyway/blob/master/flyway-core/src/main/java/org/flywaydb/core/internal/resolver/sql/SqlMigrationResolver.java#L151